src/pages/home/ReportScreen.js

```diff
@@ -276,111 +276,107 @@ class ReportScreen extends React.Component {
+                                <View
+                                    style={[styles.flex1, styles.justifyContentEnd, styles.overflowHidden]}
+                                    onLayout={(event) => {
+                                        // Rounding this value for comparison because they can look like this: 411.9999694824219
+                                        const skeletonViewContainerHeight = Math.round(event.nativeEvent.layout.height);
+
+                                        // Only set state when the height changes to avoid unnecessary renders
+                                        if (reportActionsListViewHeight === skeletonViewContainerHeight) return;
+
+                                        // The height can be 0 if the component unmounts - we are not interested in this value and want to know how much space it
+                                        // takes up so we can set the skeleton view container height.
+                                        if (skeletonViewContainerHeight === 0) {
+                                            return;
+                                        }
+                                        reportActionsListViewHeight = skeletonViewContainerHeight;
+                                        this.setState({skeletonViewContainerHeight});
+                                    }}
+                                >
+                                    {this.isReportReadyForDisplay() && !isLoadingInitialReportActions && !isLoading && (
+                                        <ReportActionsView
                                             reportActions={this.props.reportActions}
                                             report={this.props.report}
                                             isComposerFullSize={this.props.isComposerFullSize}
-                                            onSubmitComment={this.onSubmitComment}
-                                            policies={this.props.policies}
+                                            policy={policy}
                                         />
-                                    </>
-                                )}
+                                    )}
 
-                                {!this.isReportReadyForDisplay() && (
-                                    <ReportFooter
-                                        shouldDisableCompose
-                                        isOffline={this.props.network.isOffline}
-                                    />
-                                )}
-                            </View>
-                        </DragAndDropProvider>
-                    </FullPageNotFoundView>
-                </ScreenWrapper>
-            </ReportScreenContext.Provider>
+                                    {/* Note: The report should be allowed to mount even if the initial report actions are not loaded. If we prevent rendering the report while they are loading then
```

Comment:
> The report should be allowed to mount

Can you please clarify in this comment what "the report" is? Is it:
- ReportActionsSkeletonView
- ReportFooter
- ReportFooter
- All three?
- Something else?

Reply:
All these are part of linting (the component it is now wrapped into a new context provider), I have not changed or touched the internal logic. I can do it if you want, but it falls out of the scope of the performance improvements

Reply:
Aren't you lucky that you stumbled into this code? 😀 

I guess it's not a blocker for this particular PR, but I am always happy to see it when people leave the code better than how they found it.

Reply:
ok, I will take care of it then :)